presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -444,4 +444,13 @@ default void revokeTablePrivileges(ConnectorSession session, SchemaTableName tab
     {
         return emptyList();
     }
+
+    /**
+     * @return whether the connector supports not null semantics in columns
+     * @see ColumnMetadata#isNullable
+     */
+    default boolean supportsNotNullColumns()
```

Comment:
To me this should not belong here because it not metadata kind of information. It is more like connector capabilities, static and constant during the lifetime of connector. Maybe you should extract new connector entity like `ConnectorCapabilities`? Or this should be moved to `com.facebook.presto.spi.connector.Connector` like `com.facebook.presto.spi.connector.Connector#isSingleStatementWritesOnly`.

Also I would change the name of this method, maybe: `supportsNonNullableColumns`?

Reply:
CC: @electrum for discussion.

This seems like a reasonable idea, however as of yet it's not clear to me how to obtain the `Connector` (or if that's even desirable) inside `com.facebook.presto.execution`.  It is clear however how to just refactor this so that `ConnectorMetadata` has a `getConnectorCapabilities` method.

I wavered between `NonNull` and `NotNull`, but decided `NotNull` because that is the same wording as the SQL.

Reply:
My main question related to your feedback is what is the best way to seed a `ConnectorCapabilities` through to the subclasses of `DataDefinitionTask`.  By looking at the code I'm not seeing an obvious path to provide this to these classes.

Reply:
I can see both sides. Putting in `ConnectorMetadata` is nice because it's metadata related. We have `isSingleStatementWritesOnly()` in `Connector` because that is transaction related and all transaction calls are there. On the other hand, `ConnectorMetadata` is an instance per-transaction, which is not needed for this method.

I'm inclined to say we should move this to `Connector`. If we do that, we can add this as a flag to `CatalogMetadata` (created from `Catalog` which has a reference to `Conector`), which is accessible in `MetadataManager`.

Related, we have `comment` in `ColumnMetadata`, but many connectors silently ignore it. I'm thinking we should add the same behavior for comments, or, alternatively, leave this entirely up to connectors (though this seems worse).

Reply:
Thank you, I can see how to wire this, and I think it's a clean approach.

It looks like only Hive supports comments at the moment.  I'll add this support for Hive, and add a similar check in create/add column tasks, as a separate follow-up PR once this is merged.